### PR TITLE
[W4.5c][W14-A1] K Muruges

### PR DESCRIPTION
**Pull request by muruges95**
Added catch for situation when storage file become read-only whilst the program is running

### CODE REVIEW

File: src/seedu/addressbook/Main.java [hunk 1]
```diff
@@ -111,6 +112,9 @@ private CommandResult executeCommand(Command command)  {
             CommandResult result = command.execute();
             storage.save(addressBook);
             return result;
+        } catch (StorageOperationException e) {
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:115]: In general, it is good to handle the exception closer to where it occurs.

File: src/seedu/addressbook/Main.java [hunk 2]
```diff
@@ -111,6 +112,9 @@ private CommandResult executeCommand(Command command)  {
...
+            ui.showToUser(Messages.MESSAGE_STORAGE_FILE_CANNOT_OPEN);
```
Comment (okkhoy) [inline src/seedu/addressbook/Main.java:116]: In general, just showing the error message or printing the stack trace is not considered "good exception handling".
Think how you can improve the user experience in light of the exception that just occurred.

File: src/seedu/addressbook/common/Messages.java [hunk 3]
```diff
@@ -15,4 +15,5 @@
             "java seedu.addressbook.Main [STORAGE_FILE_PATH]";
     public static final String MESSAGE_WELCOME = "Welcome to your Address Book!";
     public static final String MESSAGE_USING_STORAGE_FILE = "Using storage file : %1$s";
+    public static final String MESSAGE_STORAGE_FILE_CANNOT_OPEN = "An error has occurred: The storage file is currently read-only.";
```
Comment (okkhoy) [inline src/seedu/addressbook/common/Messages.java:18]: Good to see you have segregated the message to where it belongs!

**Comment by okkhoy** [comment]
@muruges95 
Some comments added. Please ack & close the PR after reading comments.

**Comment by muruges95** [comment]
Alright roger will take note of these

**Pull request closed without merging**